pageserver/src/timelines.rs

```diff
@@ -189,15 +168,45 @@ pub fn init_pageserver(
     Ok(())
 }
 
+pub enum CreateRepo {
+    Real {
+        wal_redo_manager: Arc<dyn WalRedoManager + Send + Sync>,
+        remote_index: Arc<tokio::sync::RwLock<RemoteTimelineIndex>>,
```

Comment:
`Arc<tokio::sync::RwLock<RemoteTimelineIndex>>`

IMO, that could be a newtype like `Index()` or a `pub type Index = ` for simplicity, that would hide tokio and our hacks around better, we can even provide a sync api for the parts that we need.
Another non-big follow-up task, I guess.